### PR TITLE
Revert "Revert "Unused Blocks""

### PR DESCRIPTION
**Pull request by Hamms**
Reverts code-dot-org/code-dot-org#8712

### CODE REVIEW

**Comment by Hamms** [comment]
@bcjordan if you could take a look specifically at https://github.com/code-dot-org/code-dot-org/pull/8725/commits/9c85669d4879a5557c04e42b3650901f5858968a

File: apps/src/StudioApp.js [hunk 1]
```diff
@@ -1813,6 +1816,23 @@ function runButtonClickWrapper(callback) {
...
+    // the event constructor, but some older browsers do not yet support
+    // event constructors. Attempt to use the new functionality, fall
+    // back to the old if it fails.
+    try {
+      customEvent = new Event(Blockly.BlockSpace.EVENTS.RUN_BUTTON_CLICKED);
+    } catch (e) {
+      customEvent = document.createEvent('Event');
+      customEvent.initEvent(Blockly.BlockSpace.EVENTS.RUN_BUTTON_CLICKED, true, false);
+    }
+    Blockly.mainBlockSpace.getCanvas().dispatchEvent(customEvent);
```
Comment (bcjordan) [inline apps/src/StudioApp.js:1823]: Fix LGTM!

One nit, can take or leave, feels like maybe a tiny bit too much logic living in this portion of the code. Couple of ideas:
- add as a `./utils.js` method, so would call e.g., `utils.createEvent(Blockly.BlockSpace.EVENTS.RUN_BUTTON_CLICKED)`
- if something in Google Closure tools / blockly utils supports this sort of dispatching and deals with the x-browser issue under the hood (do other events get triggered similarly?), another option could be to move this portion of the logic down in to blockly-core and use a closure tools helper. So we then might call e.g., `Blockly.mainBlockSpace.dispatchRunButtonClicked();`

Comment (bcjordan) [inline apps/src/StudioApp.js:1823]: Also perfectly reasonable to skip if you've done barrage of x-browser tests on this and would prefer to hold off on doing another just to move a bit of logic around 😄

Comment (Hamms) [inline apps/src/StudioApp.js:1823]: Good call!

File: apps/src/StudioApp.js [hunk 2]
```diff
@@ -1819,17 +1819,7 @@ function runButtonClickWrapper(callback) {
...
-    // the event constructor, but some older browsers do not yet support
-    // event constructors. Attempt to use the new functionality, fall
-    // back to the old if it fails.
-    try {
-      customEvent = new Event(Blockly.BlockSpace.EVENTS.RUN_BUTTON_CLICKED);
-    } catch (e) {
-      customEvent = document.createEvent('Event');
-      customEvent.initEvent(Blockly.BlockSpace.EVENTS.RUN_BUTTON_CLICKED, true, false);
-    }
+    var customEvent = utils.createEvent(Blockly.BlockSpace.EVENTS.RUN_BUTTON_CLICKED);
```
Comment (bcjordan) [inline apps/src/StudioApp.js:1822]: @Hamms in this commit the first bubbles parameter in this usage changed from `true` -> `false` (default)

Comment (Hamms) [inline apps/src/StudioApp.js:1822]: it did! I realized that bubbling was unnecessary for this use case, as evidenced by the fact that the `new Event` approach works fine, and bubbling defaults to false in that case.

**Comment by bcjordan** [comment]
LGTM!

**Pull request merged**